src/main/java/com/google/api/generator/engine/ast/GeneralForStatement.java

```diff
@@ -61,22 +65,24 @@ public static Builder builder() {
 
   @AutoValue.Builder
   public abstract static class Builder {
-    public abstract Builder setInitializationExpr(Expr initializationExpr);
-
-    public abstract Builder setBody(List<Statement> body);
-
-    // Private.
-    abstract Builder setLocalVariableExpr(VariableExpr variableExpr);
-
-    abstract Builder setMaxSizeExpr(Expr maxSizeExpr);
-
-    abstract VariableExpr localVariableExpr();
+    // Private setter.
```

Comment:
Are we trying to make these private, or just visible for testing? In the first case, this Builder class should be "private" too, and we wouldn't test the other methods. Otherwise, it should be fine to expose these.

Reply:
We should try to make these private setter at this stage, only expose convenience wrapper to user.  My thought is to check type for initialization, termination, increment expression. So I also try to add unit test for the type-checking. I am not sure which one to choose, need your opinion:
1. if we make Builder private,  and we wouldn't test the methods,  do we still need to check the expressions' types?
2. or we make these tests VisibleForTesting